crucible-go fails to compile after FunctionBindings became a newtype

As of commit 049000a567d69d3d24e9c108dec17fe04e8bc29c, `crucible-go` fails to compile with the following error:

```
$ cabal build crucible-go
Build profile: -w ghc-8.10.4 -O1
In order, the following will be built (use -v for more details):
 - crucible-go-0.1.0.0 (lib) (first run)
 - crucible-go-0.1.0.0 (exe:crux-go) (first run)
Preprocessing library for crucible-go-0.1.0.0..
Building library for crucible-go-0.1.0.0..
[7 of 7] Compiling Lang.Crucible.Go.Simulate ( src/Lang/Crucible/Go/Simulate.hs, /home/rscott/Documents/Hacking/Haskell/crucible/dist-newstyle/build/x86_64-linux/ghc-8.10.4/crucible-go-0.1.0.0/build/Lang/Crucible/Go/Simulate.o, /home/rscott/Documents/Hacking/Haskell/crucible/dist-newstyle/build/x86_64-linux/ghc-8.10.4/crucible-go-0.1.0.0/build/Lang/Crucible/Go/Simulate.dyn_o )

src/Lang/Crucible/Go/Simulate.hs:92:36: error:
    • Couldn't match expected type ‘FunctionBindings p sym ext’
                  with actual type ‘FnHandleMap f0’
    • In the expression: emptyHandleMap
      In an equation for ‘mkFunctionBindings’:
          mkFunctionBindings _overrides [] = emptyHandleMap
    • Relevant bindings include
        _overrides :: [SomeOverride p sym ext]
          (bound at src/Lang/Crucible/Go/Simulate.hs:92:20)
        mkFunctionBindings :: [SomeOverride p sym ext]
                              -> [(Maybe (Text, FunctionName), AnyCFG ext)]
                              -> FunctionBindings p sym ext
          (bound at src/Lang/Crucible/Go/Simulate.hs:92:1)
   |
92 | mkFunctionBindings _overrides [] = emptyHandleMap
   |                                    ^^^^^^^^^^^^^^

src/Lang/Crucible/Go/Simulate.hs:98:39: error:
    • Couldn't match type ‘sym0’ with ‘sym’
        ‘sym0’ is untouchable
          inside the constraints: ret1 ~ ret
          bound by a pattern with constructor:
                     Refl :: forall k (a :: k). a :~: a,
                   in a case alternative
          at src/Lang/Crucible/Go/Simulate.hs:98:30-33
      ‘sym’ is a rigid type variable bound by
        the type signature for:
          mkFunctionBindings :: forall sym p ext.
                                [SomeOverride p sym ext]
                                -> [(Maybe (Text, FunctionName), AnyCFG ext)]
                                -> FunctionBindings p sym ext
        at src/Lang/Crucible/Go/Simulate.hs:(88,1)-(91,48)
      Expected type: FnState p0 sym0 ext init ret
        Actual type: FnState p sym ext args ret1
    • In the expression: UseOverride override
      In a case alternative:
          (Just Refl, Just Refl) -> UseOverride override
      In the expression:
        case
            (testEquality (cfgArgTypes cfg) argsRepr, 
             testEquality (cfgReturnType cfg) retRepr)
        of
          (Just Refl, Just Refl) -> UseOverride override
          _ -> panic
                 "[Go simulator] mkFunctionBindings"
                 ["Type mismatch for override of " ++ show ident,
                  "  Expected: "
                    ++ show (cfgArgTypes cfg) ++ " -> " ++ show (cfgReturnType cfg),
                  ....]
    • Relevant bindings include
        override :: Override p sym ext args ret1
          (bound at src/Lang/Crucible/Go/Simulate.hs:95:50)
        f :: FnState p0 sym0 ext init ret
          (bound at src/Lang/Crucible/Go/Simulate.hs:94:7)
        overrides :: [SomeOverride p sym ext]
          (bound at src/Lang/Crucible/Go/Simulate.hs:93:20)
        mkFunctionBindings :: [SomeOverride p sym ext]
                              -> [(Maybe (Text, FunctionName), AnyCFG ext)]
                              -> FunctionBindings p sym ext
          (bound at src/Lang/Crucible/Go/Simulate.hs:92:1)
   |
98 |             (Just Refl, Just Refl) -> UseOverride override
   |                                       ^^^^^^^^^^^^^^^^^^^^

src/Lang/Crucible/Go/Simulate.hs:105:5: error:
    • Couldn't match expected type ‘FunctionBindings p sym ext’
                  with actual type ‘FnHandleMap (FnState p0 sym0 ext)’
    • In the expression:
        insertHandleMap (cfgHandle cfg) f
          $ mkFunctionBindings overrides cfgs
      In the expression:
        let
          f = case lookupOverride' ident overrides of
                Just (SomeOverride _pkg argsRepr retRepr override) -> ...
                Nothing -> ...
        in
          insertHandleMap (cfgHandle cfg) f
            $ mkFunctionBindings overrides cfgs
      In an equation for ‘mkFunctionBindings’:
          mkFunctionBindings overrides ((ident, AnyCFG cfg) : cfgs)
            = let f = ...
              in
                insertHandleMap (cfgHandle cfg) f
                  $ mkFunctionBindings overrides cfgs
    • Relevant bindings include
        f :: FnState p0 sym0 ext init ret
          (bound at src/Lang/Crucible/Go/Simulate.hs:94:7)
        cfgs :: [(Maybe (Text, FunctionName), AnyCFG ext)]
          (bound at src/Lang/Crucible/Go/Simulate.hs:93:53)
        cfg :: CFG ext blocks init ret
          (bound at src/Lang/Crucible/Go/Simulate.hs:93:46)
        overrides :: [SomeOverride p sym ext]
          (bound at src/Lang/Crucible/Go/Simulate.hs:93:20)
        mkFunctionBindings :: [SomeOverride p sym ext]
                              -> [(Maybe (Text, FunctionName), AnyCFG ext)]
                              -> FunctionBindings p sym ext
          (bound at src/Lang/Crucible/Go/Simulate.hs:92:1)
    |
105 |     insertHandleMap (cfgHandle cfg) f $ mkFunctionBindings overrides cfgs
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/Lang/Crucible/Go/Simulate.hs:105:41: error:
    • Couldn't match expected type ‘FnHandleMap (FnState p0 sym0 ext)’
                  with actual type ‘FunctionBindings p sym ext’
    • In the second argument of ‘($)’, namely
        ‘mkFunctionBindings overrides cfgs’
      In the expression:
        insertHandleMap (cfgHandle cfg) f
          $ mkFunctionBindings overrides cfgs
      In the expression:
        let
          f = case lookupOverride' ident overrides of
                Just (SomeOverride _pkg argsRepr retRepr override) -> ...
                Nothing -> ...
        in
          insertHandleMap (cfgHandle cfg) f
            $ mkFunctionBindings overrides cfgs
    • Relevant bindings include
        f :: FnState p0 sym0 ext init ret
          (bound at src/Lang/Crucible/Go/Simulate.hs:94:7)
        cfgs :: [(Maybe (Text, FunctionName), AnyCFG ext)]
          (bound at src/Lang/Crucible/Go/Simulate.hs:93:53)
        cfg :: CFG ext blocks init ret
          (bound at src/Lang/Crucible/Go/Simulate.hs:93:46)
        overrides :: [SomeOverride p sym ext]
          (bound at src/Lang/Crucible/Go/Simulate.hs:93:20)
        mkFunctionBindings :: [SomeOverride p sym ext]
                              -> [(Maybe (Text, FunctionName), AnyCFG ext)]
                              -> FunctionBindings p sym ext
          (bound at src/Lang/Crucible/Go/Simulate.hs:92:1)
    |
105 |     insertHandleMap (cfgHandle cfg) f $ mkFunctionBindings overrides cfgs
    |                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
cabal: Failed to build crucible-go-0.1.0.0 (which is required by exe:crux-go
from crucible-go-0.1.0.0).
```

--- a/crucible-go/src/Lang/Crucible/Go/Simulate.hs
+++ b/crucible-go/src/Lang/Crucible/Go/Simulate.hs
@@ -89,7 +89,7 @@ mkFunctionBindings :: forall sym p ext.
                       [SomeOverride p sym ext]
                    -> [(Maybe (Text, FunctionName), AnyCFG ext)]
                    -> FunctionBindings p sym ext
-mkFunctionBindings _overrides [] = emptyHandleMap
+mkFunctionBindings _overrides [] = FnBindings emptyHandleMap
 mkFunctionBindings overrides ((ident, AnyCFG cfg) : cfgs) =
   let f = case lookupOverride' ident overrides of
         Just (SomeOverride _pkg argsRepr retRepr override) ->
@@ -102,7 +102,9 @@ mkFunctionBindings overrides ((ident, AnyCFG cfg) : cfgs) =
                  , "  Got: " ++ show argsRepr ++ " -> " ++ show retRepr
                  ]
         Nothing -> UseCFG cfg $ C.postdomInfo cfg in
-    insertHandleMap (cfgHandle cfg) f $ mkFunctionBindings overrides cfgs
+    FnBindings $ insertHandleMap (cfgHandle cfg) f
+               $ fnBindings
+               $ mkFunctionBindings overrides cfgs
 
 asApp :: MonadFail m
       => Reg.Expr ext s tp

